src/main/java/hudson/plugins/sonar/SonarGlobalConfiguration.java

```diff
@@ -72,25 +73,20 @@ public void setBuildWrapperEnabled(boolean enabled) {
   }
 
   /**
-   * Attempts to migrated data from SonarPublished, which was previously holding the global configuration.
-   * It is thread safe and will refuse to migrate if a SonarQube installation already exists in this class.
+   * Attempts to migrate data from SonarPublished, which was previously holding the global configuration.
+   * It will refuse to migrate if a SonarQube installation already exists in this class.
    * Migration will only be attempted once. 
    */
-  @Initializer(after = InitMilestone.PLUGINS_PREPARED)
-  public void migrate() {
-    if (migrated) {
+  @SuppressWarnings("deprecation")
+  @Initializer(after = InitMilestone.JOB_LOADED)
+  public void migrateData() {
+    if (dataMigrated) {
       return;
     }
-
-    synchronized (this) {
-      if (migrated) {
-        return;
-      }
-      // SonarPublisher might be null if Maven plugin is disabled or not installed
-      Jenkins j = Jenkins.getInstance();
-      DescriptorImpl publisher = j.getDescriptorByType(SonarPublisher.DescriptorImpl.class);
-      if (publisher != null && publisher.getDeprecatedInstallations() != null && publisher.getDeprecatedInstallations().length > 0) {
-
+    Optional<DescriptorImpl> publisherOpt = ExtensionList.lookup(SonarPublisher.DescriptorImpl.class).stream().findFirst();
+    // SonarPublisher might be missing if Maven plugin is disabled or not installed
+    publisherOpt.ifPresent(publisher -> {
+      if (publisher.getDeprecatedInstallations() != null && publisher.getDeprecatedInstallations().length > 0) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "getDeprecatedInstallations"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=91&issues=AWKrOeFbUMN1uZ1OXWHY&open=AWKrOeFbUMN1uZ1OXWHY)

[ISSUE_KEY]: AWKrOeFbUMN1uZ1OXWHY